From b4ba270c27620cde1e76f4dce74fc7678fe268fd Mon Sep 17 00:00:00 2001
From: moinakde <moinak.debnath@intel.com>
Date: Mon, 24 Jun 2019 08:42:06 +0530
Subject: [PATCH] UGW_SW-39867 : EoGRE LAN Terminaton

New session flags for EoGRE ingress and egress sessions.
New exported api for EoGRE inner dmac address.
---
 include/net/ppa/ppa_stack_al.h | 10 ++++++++++
 include/uapi/net/ppa_api.h     |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/include/net/ppa/ppa_stack_al.h b/include/net/ppa/ppa_stack_al.h
index ae0b90cd104e..c174325ca106 100644
--- a/include/net/ppa/ppa_stack_al.h
+++ b/include/net/ppa/ppa_stack_al.h
@@ -1655,6 +1655,16 @@ int32_t ppa_form_gre_hdr(PPA_NETIF *dev,
 		uint16_t dataLen,
 		uint8_t *pHdr,
 		uint16_t *len);
+/*! \brief This function returns EOGRE inner destination mac address.
+	\param[out] mac Pointer to destination mac address.
+	\param[in] skb Pointer to the packet buffer.
+	\param[in] dst Pointer to the route table entry.
+	\return Returns PPA_SUCCESS value if GRE inner dmac is formed.
+	\note
+ */
+int32_t ppa_get_dmac_from_dst_entry( uint8_t* mac,
+		PPA_BUF* skb,
+		struct dst_entry *dst);
 /*! \brief This function performs a bridge forwarding database lookup for the bridge specified by netif and returns the member interface on which the packet needs to be forwarded.
 	\param[in] netif Pointer to the network interface structure for the bridge interface where destination lookup is to be performed.
 	\param[in] mac Pointer to destination mac address.
diff --git a/include/uapi/net/ppa_api.h b/include/uapi/net/ppa_api.h
index 1c98461736aa..6da345ddaacf 100755
--- a/include/uapi/net/ppa_api.h
+++ b/include/uapi/net/ppa_api.h
@@ -487,6 +487,8 @@
 #define SESSION_FLAG2_NON_ETHWAN_SESSION 	0x00010000
 #define SESSION_FLAG2_DS_MPE_QOS		0x00020000 /* Flag for DS_QOS capability */
 #define SESSION_FLAG2_VALID_IPSEC_TRANS		0x00040000 /* Flag for L2TP over IPSec Transport mode */
+#define SESSION_FLAG2_IG_GRE			0x00080000 /* Flag for Ingress GRE session*/
+#define SESSION_FLAG2_EG_GRE			0x00100000 /* Flag for Engress GRE session*/
 
 /* Other flags */
 #define FLG_PPA_PROCESSED		0x100	/* this used to mark ecah packets which are processed by ppa datapath driver*/
